iOS/Sources/Beagle/Sources/Navigation/BeagleNavigator.swift

```diff
@@ -296,4 +296,24 @@ class BeagleNavigator: BeagleNavigation {
             }
         }
     }
+    
+    private func transformAdditionalDataObject(
+        from path: Route.NewPath,
+        with origin: UIView?
+    ) -> HttpAdditionalData {
+        guard let method = path.httpAdditionalData?.method,
+              let body = path.httpAdditionalData?.body else {
+            return HttpAdditionalData(httpData: nil, headers: path.httpAdditionalData?.headers ?? [:])
+        }
+        
+        let encoder = JSONEncoder()
+        let bodyEvaluated = body.evaluate(with: origin)
+        let data: Data? = try? encoder.encode(bodyEvaluated)
+        return HttpAdditionalData(
+            httpData: .init(method: method.toMethod() ?? .GET,
+                            body: data ?? Data()),
+            headers: path.httpAdditionalData?.headers ?? [:]
+        )
+        
+    }
```

Comment:
`guard let` both `body` and `method` will force the method GET when no body is provided (i.e. a DELETE request).
```suggestion
    private func transformAdditionalDataObject(
        from path: Route.NewPath,
        with origin: UIView?
    ) -> HttpAdditionalData {
        let body: Data?
        let method = path.httpAdditionalData?.method?.toMethod()
        if let evaluated = path.httpAdditionalData?.body?.evaluate(with: origin) {
            body = try? JSONEncoder().encode(evaluated)
        } else {
            body = nil
        }
        return HttpAdditionalData(
            httpData: .init(
                method: method ?? .GET,
                body: body ?? Data()
            ),
            headers: path.httpAdditionalData?.headers ?? [:]
        )
    }
```